src/main/scala/com/example/solution/domain/user.scala

```diff
@@ -0,0 +1,33 @@
+package com.example.solution.domain
+
+object user {
+
+
+  case class UserName(value: String) extends AnyVal
+
+  case class UserSurname(value: String) extends AnyVal
+
+  sealed abstract case class User private (
+                           name: UserName,
+                           surname: UserSurname
+                         )
+
+  object User {
+    def mkUser(username: UserName, userSurname: UserSurname): Option[User] = {
+      val (name, surname) = (username.value, userSurname.value)
+      if (name.length < 3 || surname.length < 3) None
```

Comment:
you choose to place validations here. It can be also moved to UserName and UserSurname. What are pros and cons of the second approach? Why you didn't choose it?

Reply:
I guess putting validation in UserName/UserSurname would be more appropriate, validation of UserName is independent from validation on UserSurname. We would need a method like map2 to create the User instance though, which I believe is not present in standard library, so we need to rely on Cats library or create our own functions.